yb-thirdparty-common.sh

```diff
@@ -107,6 +107,42 @@ activate_virtualenv() {
   )
 }
 
+# Re-executes the current script with the correct macOS architecture.
+ensure_correct_mac_architecture() {
+  if [[ $OSTYPE != darwin* ]]; then
+    return
+  fi
+  if [[ -z ${YB_TARGET_ARCH:-} ]]; then
+    local uname_output
+    uname_output=$( uname -a )
```

Comment:
Should we check more specific option instead of the "all" option?  uname -p ?

Reply:
Updated to use `uname -p`.